frontend/src/query_builder/QueryVisualization.jsx

```diff
@@ -188,6 +188,11 @@ export default class QueryVisualization extends Component {
         }
     }
 
+    showDetailError() {
```

Comment:
Shouldn't we just have a `showDetailError` state property that's used to show/hide the error?

If not then you should use `React.findDOMNode()` instead of `getDOMNode`, which is deprecated.